libs/membership/website/src/lib/has-unpaid-invoices.tsx

```diff
@@ -0,0 +1,16 @@
+import {useUser} from '@wepublish/authentication/website'
+import {useInvoicesQuery} from '@wepublish/website/api'
+import {useMemo} from 'react'
+
+export const useHasUnpaidInvoices = () => {
+  const {hasUser} = useUser()
+  const {data} = useInvoicesQuery({
+    fetchPolicy: 'cache-first',
+    skip: !hasUser
+  })
+
+  return useMemo(
+    () => data?.invoices.some(invoice => !invoice.canceledAt && !invoice.paidAt),
```

Comment:
same logic used all over the code. we could start sharing a re-usable function for that ;)

Reply:
This is the re-usable function for that.

Reply:
I can live with it. In my code I just try to make life easier / safer for the following. So, that my fellows don't need to care about the fact, that he has to check for canceledAt and paidAt. I would probably just go with something like this:

`export function isActiveInvoice (invoice: Invoice): boolean => !invoice.canceledAt && !invoice.paidAt`

This is maybe because I am coming from OOP, where you would have such a function within the Invoice class. Defined it once, used it everywhere with exact same logic.